src/kernel-install/50-depmod.install

```diff
@@ -6,6 +6,7 @@
 
 case "$1" in
     add)
+        [[ -d /lib/modules/"$2" ]] || exit 0
```

Comment:
this check doesn't really work on distros that place the kernel image in /lib/modules/.

Fedora places the kernel there, under control of the package manager, and only copies it to the boot partition from there.

Reply:
Do fedora support kernels without modules?

Reply:
fedora might not, but regardless what fedora does or not I am very sure dropping the kernel to /lib/modules alongside the modules is a very good idea, and we should suppotr that.

Reply:
I'm not sure I follow... It's definitely still supported with this patch.

The only thing this patch does is skip the "depmod" call if the /lib/modules/$2 directory doesn't exist.

Currently, when your kernel doesn't support modules, this directory doesn't exist and kernel-install fails because depmod fails

Reply:
Fedora installs the kernel image to `/lib/modules/$(uname -r)/vmlinuz` and installs it to /boot only in RPM posinst scripts. That's a good idea, because it monopolizes the OS in its entirety in /usr, which means /usr contains everything you need to boot the system properly.

The fedora model is hence a good model, one we should support as the best approach. Now in that model /lib/modules/$(uname -r) will exist regardless if the kernel is built with modules on or modules off. Hence the test you proposed doesn't really solve the problem properly, as systems following fedora's logic but which decide to turn off modules won't be covered by your check. The check should hence be more precise, and *actually* check if there are modules in place. For example checking for  `/lib/modules/$(uname -r)/kernel/` should be good enough already

Reply:
Right, I see what you mean now, will fix